vis_gen.py

```diff
@@ -0,0 +1,565 @@
+        """Read code points from file and visualize. Character set file must
+        follow this format:
+            1. Each line represents a single code point
+            2. Each code point is in format 'U+2a665'
+
+        Args:
+            file_path: Str, path to file for the character set.
+            font_styles: list of Str, styles of the font face to visualize
+            antialiases: list of Str, antialiasing styles to visualize
+        """
+        # Check if out_dir exists and create if not
+        out_dir_abs = os.path.join(os.getcwd(), self.out_dir)
+        if self.__check_out_dir:
+            self.__check_create_out_dir_abs(out_dir_abs)
+        self.__check_out_dir = False
+
+        # Read file and translate format
+        code_points = []
+        with open(file_path) as f:
+            for line in f:
+                code_point = chr(int('0x' + line.split('\n')[0][2:], 16))
+                code_points.append(code_point)
+
+        self.generate_dataset_from_list(code_points, font_styles, antialiases)
+
+
+    def generate_dataset_from_list(self, code_points, font_styles, antialiases):
+        """Render text images in the given list of code points and write to
+        out_dir.
+
+        Args:
+            code_points: list of Unicode code points to visualize
+            font_styles: list of Str, styles of the font face to visualize
+            antialiases: list of Str, antialiasing styles to visualize
+
+        Raises:
+            OSError: if specified directory cannot be created
+            TypeError: if code point format is not correct
+        """
+        # Check if out_dir exists and create if not
+        out_dir_abs = os.path.join(os.getcwd(), self.out_dir)
+        if self.__check_out_dir:
+            self.__check_create_out_dir_abs(out_dir_abs)
+        self.__check_out_dir = False
+
+        for font_style in font_styles:
+            self.font_style = font_style
+            print('Successfully selected font style: {}.'.format(font_style))
+            for antialias in antialiases:
+                self.antialias = antialias
+                print('Successfully selected antialiasing style: {}.'.format(
+                    antialias))
+                self.visualize_list(code_points)
+
+        self.__check_out_dir = True
+
+
+    def visualize_list(self, code_points, x=None, y=None):
+        """Render text images from start code point to end code point and write
+        to out_dir.
+
+        Args:
+            code_points: List of Str, a list of all codepoints to visualize
+            x: Int, x coordinate of the position of text, in number of pixels
+            y: Int, y coordinate of the position of text, in number of pixels
+
+        Raises:
+            OSError: if specified directory cannot be created
+            TypeError: if code point format is not correct
+        """
+        # Check if out_dir exists and create if not
```

Comment:
It looks like you use the following 4 lines a lot. How about wrap that into a private function instead.